File: 01 solved.ac (baekjoon)/03. Gold/q2467.java
import java.io.*;
import java.util.*;

public class q2467 {
    
    public static void main(String[] args) throws IOException {

        // (1) 입력 및 전처리
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int count = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        int[] nums = new int[count];
        for (int i = 0; i < count; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        // (2) processing
        int left = 0;
        int right = count - 1;
        int minSum = Integer.MAX_VALUE;
        int[] result = new int[2];
        while (left < right) {

            int sum = nums[left] + nums[right];
            if (Math.abs(sum) < minSum) {
                minSum = Math.abs(sum);
                result[0] = nums[left];
                result[1] = nums[right];

                if (minSum == 0) {
                    break;
                }
            }
            
            if (sum < 0) {
                left++;
            } else {
                right--;
            }

        }

        System.out.println(result[0] + " " + result[1]);
    }
}

File: 01 solved.ac (baekjoon)/03. Gold/q2467.py
"""백준 2467"""

def main() :
    """백준 2467 풀이"""
    count = int(input())
    nums = list(map(int, (input().split())))

    left = 0
    right = count - 1
    result = (0, 0)
    min_sum = float('inf')

    while left < right :
        cur_sum = nums[left] + nums[right]

        if abs(cur_sum) < min_sum :
            result = (nums[left], nums[right])
            min_sum = abs(cur_sum)
            if min_sum == 0 :
                break

        if cur_sum < 0 :
            left += 1
        else :
            right -= 1

    print(*result)

if __name__ == "__main__" :
    main()
